src/uct/api/uct.h

```diff
@@ -1140,6 +1144,10 @@ ucs_status_t uct_md_mem_dereg(uct_md_h md, uct_mem_h memh);
  * exhausted. In this case the next allocation method from the initial list will
  * be attempted.
  *
+ * @param [in]     addr        If value is not NULL, then the value will be used
```

Comment:
it would be good to clarify when addr is interpreted as a hint and when it is interpreted as a requirement.  i've attempted to do that in the following text.  please modify as needed for accuracy:

 If addr is NULL, the underlying allocation routine will choose the address at which to create the mapping.  If addr is non-NULL but UCT_MD_MEM_FLAG_FIXED is not set, the address will be interpreted as a hint as to where to establish the mapping.  if addr is non-NULL and UCT_MD_MEM_FLAG_FIXED is set, then the specified address is interpreted as a requirement.  In this case, if the mapping to the exact address cannot be made, the allocation request fails.